sympy/solvers/ode/single.py

```diff
@@ -991,6 +991,78 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
         return [gensol]
 
 
+class RationalRiccati(SinglePatternODESolver):
+    r"""
+    Gives all rational solutions to the first
+    order Riccati Differential Equation
+
+    .. math :: y' = b_0(x) + b_1(x) y + b_2(x) y^2
+
+    where `b_0`, `b_1` and `b_2` are rational functions of `x`
+    with `b_2 \ne 0` (`b_2 = 0` would make it a Bernoulli equation).
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Function, dsolve, checkodesol
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+
+    >>> eq = -x**4*f(x)**2 + x**3*f(x).diff(x) + x**2*f(x) + 20
+    >>> sol = dsolve(eq, hint="1st_rational_riccati")
+    >>> sol
+    Eq(f(x), (4*C1 - 5*x**9 - 4)/(x**2*(C1 + x**9 - 1)))
+    >>> checkodesol(eq, sol)
+    (True, 0)
+
+    .. [1] Riccati ODE -  https://en.wikipedia.org/wiki/Riccati_equation
+
+    .. [2] N. Thieu Vo - Rational and Algebraic Solutions of First-Order Algebraic ODEs,
+    Algorithm 11 (Pg 78) - https://www3.risc.jku.at/publications/download/risc_5387/PhDThesisThieu.pdf
+
+    .. [3] Examples taken from: Georg Grasegger, N. Thieu Vo, Franz Winkler -
+    Statistical Investigation of First-Order Algebraic ODEs and their Rational General Solutions
+    https://www3.risc.jku.at/publications/download/risc_5197/RISCReport15-19.pdf
```

Comment:
This reference seems out of place here. Maybe it belongs somewhere else.

Reply:
Should I put it in `test_riccati.py`?

Reply:
Are the examples from that reference actually used anywhere in this PR?

Reply:
Yes, in the tests for general solutions.

Reply:
I'll put it in `test_riccati.py`